Sources/StreamChat/Workers/ChannelListCleanupUpdater.swift

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+
+class ChannelListCleanupUpdater<ExtraData: ExtraDataTypes>: Worker {
+    // MARK: - Properties
+    
+    private let channelListUpdater: ChannelListUpdater<ExtraData>
+    
+    // MARK: - Init
+    
+    override init(
+        database: DatabaseContainer,
+        apiClient: APIClient
+    ) {
+        channelListUpdater = ChannelListUpdater(database: database, apiClient: apiClient)
+        super.init(
+            database: database,
+            apiClient: apiClient
+        )
+    }
+    
+    init(
+        database: DatabaseContainer,
+        apiClient: APIClient,
+        channelListUpdater: ChannelListUpdater<ExtraData>
+    ) {
+        self.channelListUpdater = channelListUpdater
+        super.init(
+            database: database,
+            apiClient: apiClient
+        )
+    }
+    
+    func cleanupChannelList() {
+        database.write { _ in
+            self.allChannels.forEach { $0.syncFailedCleanUp() }
+        } completion: { error in
```

Comment:
IMO the `self.allChannels` indirection is not necessary and dangerous since it can be called only from within the `write` block. I think inlining it would be better:
```
database.write { session in 
   let channels = (session as NSManagedObjectContext)?.fetch(ChannelDTO.allChannelsFetchRequest)
   channels.forEach { $0. syncFailedCleanUp() }
}
```